### DISCUSSION TITLE
Help needed.    Adding new Language Pack for Khmer

### DISCUSSION THREAD

**Discussion by adnanmacro** [AnySoftKeyboard/AnySoftKeyboard#3324] [Q&A]
Hi, I am trying to make a new language for kher language. I have make the required changes in the java files. 

I am facing issue while adding the dictionary. I have added the .dic files but it is now showing the suggestion. can you please guide.
2nd can you please guide me to create the .gz file for Khmer language ( is it necessary to create this file)

please assist.
![image](https://user-images.githubusercontent.com/6647618/164726568-b3dc85c8-fc93-4288-83fb-18519f82b672.png)

**Comment by sslaia** [comment]
Hi @adnanmacro the keyboard will not show suggestion, if you have cancelled it before (clicking on the x icon on the top bar, where the suggestion appears). I don't know how to get it back, but I think by clearing the cache of AnySoftKeyboard app in the system Settings, you'd be prompted with new setup of the keyboard.

To your 2nd question, I don't know what you mean. The dictionary file is a simple xml file. You can see the example from other languages in the folder addons/languages/XXX/pack/dictionary/prebuilt/

**Comment by RusneroV** [comment]
I'm not a programmer. From what I understood, for the dictionaries you need lots of text files with text for the language you want and place it in /pack/dictionary/inputs then run ./gradlew :addons:languages:your_language:pack:makeDictionary . I believe it auto creates the files needed.

**Comment by menny** [comment]
These are all correct suggestions.
Technically, there are three types of files that can be used (with or without combination) to create the pack's dictionary:
1) `gz` file under `[path/to/pack/folder/]dictionary/`. For example, English [gz[(https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/main/addons/languages/english/pack/dictionary/aosp.combined.gz). This file is an official AOSP wordlist file. I got that specific file from [here](https://github.com/LineageOS/android_packages_inputmethods_LatinIME/tree/lineage-19.1/dictionaries)
2) raw word-list xml files under `[path/to/pack/folder/]dictionary/prebuilt`. For example, some [words](https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/main/addons/languages/english/pack/dictionary/prebuilt/additionals.xml) I needed.
3) text files that are read, collected, and converted into word-lists under `[path/to/pack/folder/]dictionary/inputs`. This can be anything, but make sure it is a plain text file, nothing marked-down or structured.

**Reply by menny** [reply]
some of this is covered here: https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/main/addons/CONTRIBUTING.md
